### PR TITLE
Fixed several issues with giphy & googlesearch, and increased test coverage

### PR DESCRIPTION
**Pull request by fredfishgames**
Fixed issue where giphy posts error to console on test
Removed redundant code in googlesearch
Made googlesearch & googletranslate have 100% test coverage

### CODE REVIEW

File: zulip_bots/zulip_bots/bots/googlesearch/test_googlesearch.py [hunk 1]
```diff
@@ -28,3 +30,15 @@ def test_bot_help(self) -> None:
     def test_bot_no_results(self) -> None:
         with self.mock_http_conversation('test_no_result'):
             self.verify_reply('no res', 'Found no results.')
+
+    def test_attribute_error(self) -> None:
+        with self.mock_http_conversation('test_attribute_error'), \
+                patch('logging.exception'):
+            self.verify_reply('test', 'Error: Search failed. \'NoneType\' object has no attribute \'findAll\'.')
+
+    def test_bad_links(self) -> None:
```
Comment (showell) [inline zulip_bots/zulip_bots/bots/googlesearch/test_googlesearch.py]: This test is a bit confusing to me.  If there are bad links, why does the reply say "Found Result"?

Comment (fredfishgames) [reply zulip_bots/zulip_bots/bots/googlesearch/test_googlesearch.py]: Oh, bad links are links that should be ignored
Any cached or empty results

Comment (fredfishgames) [reply zulip_bots/zulip_bots/bots/googlesearch/test_googlesearch.py]: It makes sure it ignores those links, then sends the zulip link which is 'under' the bad ones

Comment (fredfishgames) [reply zulip_bots/zulip_bots/bots/googlesearch/test_googlesearch.py]: To test all the cases in:
```python
try:
            # Tries to get the href property of the URL
            link = a['href']
        except KeyError:
            continue
        # Link must start with '/url?', as these are the search result links
        if (not link.startswith('/url?')):
            continue
        # Makes sure a hidden 'cached' result isn't displayed
        if a.text.strip() == 'Cached' and 'webcache.googleusercontent.com' in a['href']:
            continue
        # a.text: The name of the page
        result = {'url': "https://www.google.com{}".format(link),
                  'name': a.text}
        results.append(result)
```

Comment (fredfishgames) [reply zulip_bots/zulip_bots/bots/googlesearch/test_googlesearch.py]: oh, left some brackets in

Comment (fredfishgames) [reply zulip_bots/zulip_bots/bots/googlesearch/test_googlesearch.py]: @showell Should I rename the test?

Comment (showell) [reply zulip_bots/zulip_bots/bots/googlesearch/test_googlesearch.py]: yeah, and/or clarify in a comment what's supposed to happen

**Comment by showell** [comment]
LGTM will merge when build passes

**Comment by showell** [comment]
merged!  thanks @fredfishgames!

**Comment by showell** [comment]
Oh, btw @fredfishgames just a reminder to use imperative tense in your commit messages.  So say "Fix" instead of "Fixed".  It's not a big deal (and I obviously didn't notice it when I merged), but it's our convention.

**Pull request merged**